tests/core/evm/test_evm.py

```diff
@@ -20,34 +20,44 @@
 
 
 def test_get_vm_class_for_block_number():
-    evm_class = Chain.configure(
+    chain_class = Chain.configure(
         vm_configuration=(
             (constants.GENESIS_BLOCK_NUMBER, FrontierVM),
             (constants.HOMESTEAD_MAINNET_BLOCK, HomesteadVM),
         ),
     )
-    evm = evm_class(get_db_backend(), BlockHeader(1, 0, 100))
-    assert evm.get_vm_class_for_block_number(
+    chain = chain_class(get_db_backend(), BlockHeader(1, 0, 100))
+    assert chain.get_vm_class_for_block_number(
         constants.GENESIS_BLOCK_NUMBER,) == FrontierVM
-    assert evm.get_vm_class_for_block_number(
+    assert chain.get_vm_class_for_block_number(
         constants.HOMESTEAD_MAINNET_BLOCK - 1) == FrontierVM
-    assert evm.get_vm_class_for_block_number(
+    assert chain.get_vm_class_for_block_number(
         constants.HOMESTEAD_MAINNET_BLOCK) == HomesteadVM
-    assert evm.get_vm_class_for_block_number(
+    assert chain.get_vm_class_for_block_number(
         constants.HOMESTEAD_MAINNET_BLOCK + 1) == HomesteadVM
 
 
-def test_get_vm_class_for_block_number_evm_not_found():
-    evm_class = Chain.configure(vm_configuration=())
-    evm = evm_class(get_db_backend(), BlockHeader(1, 0, 100))
-    with pytest.raises(ChainNotFound):
-        evm.get_vm_class_for_block_number(constants.GENESIS_BLOCK_NUMBER)
+def test_invalid_if_no_vm_configuration():
+    chain_class = Chain.configure(vm_configuration=())
+    with pytest.raises(ValidationError):
+        chain_class(get_db_backend(), BlockHeader(1, 0, 100))
```

Comment:
I think a ValueError or a custom exception class would be more appropriate since it's at instantiation time and not in one of the validate* methods